collector/terra/fcd/collector.go

```diff
@@ -0,0 +1,94 @@
+package fcd
+
+import (
+	"github.com/dezswap/cosmwasm-etl/pkg/db/schemas"
+	"github.com/dezswap/cosmwasm-etl/pkg/dex/terraswap"
+	"github.com/pkg/errors"
+)
+
+type fcdRepo interface {
+	TxsOf(addr string, collectedOffset, untilHeight, limit uint32) (Txs, error)
+	HasMoreTx(addr string, collectedOffset uint32) (bool, error)
+}
+
+type permanentStore interface {
+	FirstTxOf(addr string) (schemas.FcdTxLog, error)
+	Inserts(txLogs []schemas.FcdTxLog) error
+	TxLogsByHeight(height int) ([]schemas.FcdTxLog, error)
+}
+
+type ColumbusCollector interface {
+	Collect(addr string, height uint32) error
+
+	storeTxs(txs Txs) error
+	hasCollected(addr string, height uint32) (bool, error)
+
+	fcdRepo
+	permanentStore
+}
+
+type columbusFcdCollector struct {
+	fcdRepo
+	permanentStore
+}
+
+func New(repo fcdRepo, store permanentStore) ColumbusCollector {
+	return &columbusFcdCollector{repo, store}
+}
+
+// hasCollected checks the absence of the previous transaction by examining the transactions collected from the FCD server
+//
+// NOTE: fcd returns transactions in reverse order of height
+func (c *columbusFcdCollector) hasCollected(addr string, minHeight uint32) (bool, error) {
```

Comment:
whatabout deleting `minHeight`? seems not used

Reply:
https://github.com/dezswap/cosmwasm-etl/pull/15/commits/c07da70eaeb89d08fa1de79b63e592ff1769b041